### ISSUE TITLE
Use Pretzel and Github Pages for Pretzels own doc's.

### ISSUE THREAD

**Issue by biohazard999** [Code52/pretzel#350]
It would be cool if we could use [GithubPages](https://help.github.com/en/articles/configuring-a-publishing-source-for-github-pages) and pretzel it self to build the documentation.
This way it would be possible to contribute to documentation and help without the need of githubs wiki system.

Proposed solutions:
1.
- Configure GithubPages with the master branch and docs folder
- Add a pretzel site to the project under `_docs` (or something similar) and output the site to `docs`.
- Add a build step that the pretzel site will be built

2.
Use a seperate repository like explained by [Devon Burriss](https://devonburriss.me/pretezel-blog-appveyor-deployment/)

This will address #141

**Comment by laedit** [comment]
Nice idea, I wanted to improve the [site](http://code52.org/pretzel/) that is already using github pages on the gh-pages branch but haven't thought of adding the docs.

**Comment by laedit** [comment]
I agree to move the generated site source to a `docs` repo instead of having it in another branch.

It is possible to create and push a commit from AppVeyor, I already do something similar for some my sites.

**Comment by biohazard999** [comment]
> I agree to move the generated site source to a docs repo instead of having it in another branch.

Do you really want to make a repo? Or did you ment folder?

> It is possible to create and push a commit from AppVeyor, I already do something similar for some of my sites.

👍

**Comment by laedit** [comment]
> Do you really want to make a repo? Or did you ment folder?

Folder, sorry.
I haven't the rights to create another repo in Code52 so the folder should do it :)

**Comment by biohazard999** [comment]
Is github pages enabled? It kinda has to be, cause [http://code52.github.io/pretzel](http://code52.github.io/pretzel) redirects to [http://code52.org/pretzel/](http://code52.org/pretzel/)

**Comment by biohazard999** [comment]
Of course it is cause there it's https://github.com/Code52/code52.github.com
That would conflict with the idea.... Cause i felt in this trap my own cause i had an old GHP repo laying around in my profile

**Comment by biohazard999** [comment]
Which isn't built with pretzel on it's own, what an irony 😆

**Comment by laedit** [comment]
Yep it already is, using the `gh-pages` branch and the custom domain http://code52.org.
That won't conflict because https://github.com/Code52/code52.github.com handle the main site and th github page on the pretzel repo handles the code52.org/pretzel part.

Well, [github use Jekyll](https://help.github.com/en/articles/about-github-pages-and-jekyll) to generate the site so this is a good opportunity to check the equivalence with pretzel :)
And remove the need to generate the site during the AppVeyor build.

**Comment by biohazard999** [comment]
But as far as i tested if there is a organization gh-page it's not possible to use gh-pages for the repositories :/

**Comment by laedit** [comment]
Well github says that it is [possible](https://help.github.com/en/articles/user-organization-and-project-pages):
>A Project Pages site for an organization account is available at http(s)://\<orgname>.github.io/\<projectname>.

**Comment by laedit** [comment]
To be sure, we just have to modified the gh-pages branch in pretzel repo (like adding an html comment) and see if it appears on http://code52.org/pretzel/ :)

**Comment by biohazard999** [comment]
I don't even know where this comes from 🤷‍♂️ I just looked into the repo and can't find a clue where this lives or where this does come from...

**Comment by biohazard999** [comment]
[Ahhhh](https://github.com/Code52/pretzel/tree/gh-pages).

**Comment by laedit** [comment]
https://github.com/Code52/pretzel/tree/gh-pages

**Comment by laedit** [comment]
I have switched the github pages settings to use the `docs` folder in master instead of the `gh-pages` branch.

**Comment by biohazard999** [comment]
Does this work on your side? 
>Is github pages enabled? It kinda has to be, cause http://code52.github.io/pretzel redirects to http://code52.org/pretzel/
seams like it still redirects to the old page?

**Comment by laedit** [comment]
The url won't change, github page is configured to serve it to http://code52.org/pretzel/
But I can guarantee that it works, I received a mail from github to correct some old markdown parameters :)

**Comment by biohazard999** [comment]
Awesome!
Now we only start using pretzel to build the docs 👍

**Comment by laedit** [comment]
Apart if we need to use some specific plugins, since github pages use Jekyll it is a good way to check that we have parity with jekyll (at least for functionalities used by the docs).

**Comment by biohazard999** [comment]
We could do some kind of snapshot testing for that?

**Comment by laedit** [comment]
That's a good idea, but it will need a thorough followup on each jekyll release.
